src/core/stream_recv.c

```diff
@@ -942,22 +938,26 @@ QuicStreamRecvFlush(
 
         if (Status == QUIC_STATUS_CONTINUE) {
             CXPLAT_DBG_ASSERT(!Stream->Flags.SentStopSending);
-            InterlockedExchangeAdd64(
-                (int64_t*)&Stream->RecvCompletionLength,
-                (int64_t)Event.RECEIVE.TotalBufferLength);
-            FlushRecv = TRUE;
             //
             // The app has explicitly indicated it wants to continue to
             // receive callbacks, even if all the data wasn't drained.
             //
             Stream->Flags.ReceiveEnabled = TRUE;
+            if (Event.RECEIVE.TotalBufferLength == 0) {
+                continue;
+            }
+
+            InterlockedExchangeAdd64(
+                (int64_t*)&Stream->RecvCompletionLength,
+                (int64_t)Event.RECEIVE.TotalBufferLength);
+            FlushRecv = TRUE;
 
         } else if (Status == QUIC_STATUS_PENDING) {
             //
             // The app called the receive complete API inline if
             // RecvCompletionLength is non-zero.
             //
-            FlushRecv = (Stream->RecvCompletionLength != 0);
```

Comment:
This change isn't correct. We must handle the case where app calls `StreamReceiveComplete` in the callback but then returns `QUIC_STATUS_PENDING`.